---
title: 'Bulk RNAseq analysis, Class 1: Introduction to RNAseq data'
output: github_document
---

<!--class1.md is generated from class1.Rmd. Please edit that file -->

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Objectives

Welcome to Bulk RNAseq Analysis from fredhutch.io!
This course introduces you the following steps in performing RNAseq analysis:
- project organization and access via Hutch data storage and compute
- data pre-processing and quality assessment
- read mapping and summary to quantify genes/transcripts
- differential gene analysis and visualization

This course assumes a general knowledge of both the Unix shell and R statistical programming.
See [Prerequisites](../README#prerequisites) for more information about what concepts and skills should be familiar to you before this course.

For more information about RNAseq and genomics work, 
please see the resources for [RNA approaches](https://sciwiki.fredhutch.org/generation/datagen_rnaApproaches) 
in the Fred Hutch Biomedical Data Science Wiki.

Before proceeding with these training materials,
please ensure you have installed XXX as described [here](http://www.fredhutch.io/software/).

By the end of this class, you should be able to:
- Organize files (data, code, results) associated with genomics projects
- Trim and filter raw genomic data based
- Assess the quality of genomic data

## About the data

Human and mouse data from ENCODE project

FIXME diagram of study design

3x3 comparison of two tissues

Work through human materials in class,
mouse data provided and available for homework and personal exploration

For first two classes,
only work with single sample (save time).
only chromosome 9, result in sparse gene matrix

Intermediate results available for reference

Pre-analyzed data for all samples provided for third and fourth classes.

whole genome, complete count matrix for multiple samples; result is DGE across multiple samples

Section 4 onwards 
https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html

## Overview of workflow

FIXME diagram of workflow

Questions to ask at each step of the analysis:
- Are data organized and formatted appropriately?
- What characteristics of the data type and experimental design need to be considered?
- Are the results reliable, verifiable, and consistent with expectations?

## Project organization, compute environment, and reproducibility

Project-based file management

How genomics projects differ, with data stored separately

connections with Hutch infrastructure
- network drives
- accessing and transferring Unix-stored data from Mac vs Windows

Connections to reproducibility
- record keeping
- helps you in the future
- can potentially help other people as well

Logging on to `rhino`
https://sciwiki.fredhutch.org/compdemos/first_rhino/

## Trimming and filtering data

initial QC (fastqc)

trimming and quality filtering

filtering (cutadapt)
only for de novo assembly, or other special cases

follow-up QC (multiQC)

fastqc
MultiQC: https://multiqc.info

## Wrapping up

saving work

accessing work again

review objectives

preview next class's objectives

Reference materials:
https://datacarpentry.org/organization-genomics/

## On your own

## Wrapping up

**Further reading:**
- Intro to Unix
- HPC: On-premise

### Unix to cover in RNAseq skills class
- running command-line programs
- tar and gzip (and corresponding unix commands)
- idiosyncratic scripting (breaking commands onto multiple lines, flags for genomics software, etc)
