---
title: 'Bulk RNAseq analysis, Class 2: Read mapping and quantification'
output: github_document
---

<!--class2.md is generated from class2.Rmd. Please edit that file -->

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Objectives

Welcome to XXX from fredhutch.io!
This course introduces you XXX:
XXX. 

For more information about RNAseq and XXX, 
please see the [software](https://sciwiki.fredhutch.org/XXX) 
entry for the Fred Hutch Biomedical Data Science Wiki.

Before proceeding with these training materials,
please ensure you have installed XXX as described [here](http://www.fredhutch.io/software/).

By the end of this class, you should be able to:
- Use command-line bioinformatics software to map genomic reads to a reference genome
- Visualize and summarize results of read mapping
- creating unix script with all code (simple shell script sv slurm, supplemental?)
- Differentiate results for mapping genes vs transcripts? (supplemental?)
- pseudomapping? (supplemental?)
 
## Read mapping

- idiosyncratic scripting (breaking commands onto multiple lines, flags for genomics software, etc)

read mappers can handle trimming: TCGA defined two-pass workflow, can be done in a single pass

> One of the tricky parts of working in bioinformatics is that methods continue to develop rapidly.
> The incorporation of quality filtering and trimming when mapping reads is a good example.
> Understanding the assumptions of the software you're using is essential!

How to choose reference data?

connections to reference data (older index data aren't compatible with some newer programs)

## Read mapping QC

RNAseQC https://software.broadinstitute.org/cancer/cga/rna-seqc

lighterweight tool? can choose which assessments to run
IGV

## Quantifying genes

STAR2 can generate count data (stranded, unstranded) but show how to generate count data with a different tool
featureCounts from [Subread](http://subread.sourceforge.net)

> Subread is also implemented in the R package [Rsubread](http://bioconductor.org/packages/release/bioc/html/Rsubread.html)

## Reproducibiity and workflow managers

This section isn't required for this course,
but is useful to understand how analyses are scaled up for multiple samples.

Key points in reproducibility: link to ten simple rules article? distill points to those specific for genomics?

retaining commands run in scripts,
share [example shell script](https://github.com/fredhutchio/rnaseq/blob/gh-pages/skills/scripts/rnaseq.sh)

running batch jobs,
share [slurm script](https://github.com/fredhutchio/rnaseq/blob/gh-pages/skills/scripts/rnaseq.slurm)

workflow managers like Nextflow and Cromwell,
link to relevant pages in SciWiki

## Wrapping up

Review genomics command-line programs,
preview R content

## On your own
