---
title: 'Bulk RNAseq analysis, Class 2: Read mapping and quantification'
output: github_document
---

<!--class2.md is generated from class2.Rmd. Please edit that file -->

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Objectives

Welcome to XXX from fredhutch.io!
This course introduces you XXX:
XXX. 

For more information about RNAseq and XXX, 
please see the [software](https://sciwiki.fredhutch.org/XXX) 
entry for the Fred Hutch Biomedical Data Science Wiki.

Before proceeding with these training materials,
please ensure you have installed XXX as described [here](http://www.fredhutch.io/software/).

By the end of this class, you should be able to:
- Use command-line bioinformatics software to map genomic reads to a reference genome
- Visualize and summarize results of read mapping
- creating unix script with all code (simple shell script sv slurm, supplemental?)
- Differentiate results for mapping genes vs transcripts? (supplemental?)
- pseudomapping? (supplemental?)
 