# Bulk RNAseq analysis
# Class 4: Contextualizing results, and mods to RNAseq

## Objectives

By the end of this class,
you should be able to:
- functional analysis
- other considerations for analyzing RNAseq data
- other types of RNAseq data

## Functional analyses

So far we've discussed annotations in terms of gene names

We can use additional information about genes to help us understand the broader context of how genes are being differentially expressed

Connecting pathways/networks of genes, or categorizing the roles genes play: functional analysis

FIXME/IMAGE(?): for each of the following tools, provide a link and information identifying its approach to functional annotations

- GSEA
- GOseq
- GOrilla
- others?

## Other considerations for RNAseq data

FIXME/IMAGE(?): definition, example paper, etc for each of following (could duplicate content in SciWiki resource library)

- isoforms and alternative splicing
- lincRNAs
- SNVs
- RNA editing
- dealing with multiple species (xenografts, viruses)

## RNAseq beyond differential gene expression

FIXME/IMAGE(?): definition, example paper, etc for each of following (could duplicate content in SciWiki resource library)

- scRNAseq
- small RNAseq
- ribo-seq
- nascent RNAseq

## Putting it together

EXERCISE: type of analysis to interpret results (GO)

EXERCISE: want to ask a follow up question, what type of data are appropriate?

## Wrapping up

review objectives

## Errata

